flocker/apiclient/test/test_client.py

```diff
@@ -31,23 +42,22 @@ def make_clientv1_tests(client_factory, synchronize_state):
     control of this process. So when testing a real client it will be
     talking to a in-process server.
 
-    :param client_factory: 0-argument callable that returns a
-        ``IFlockerAPIV1Client`` provider.
+    The ``TestCase`` should have two 0-argument methods:
 
-    :param synchronize_state: 1-argument callable that takes a client
-        instances and makes or waits for state to match configuration.
+    create_client: Returns a ``IFlockerAPIV1Client`` provider.
+    synchronize_state: Make state match the configuration.
     """
     class InterfaceTests(TestCase):
```

Comment:
:question:  Might be nice to move the tests to a mixin in the style of the tests for other Flocker verified fakes.